packages/exchange/src/ws/integration.spec.ts

```diff
@@ -112,11 +112,47 @@ for (const host of hosts) {
                 expect(service.connectionCount).toBe(2)
 
                 service.disconnect()
+                const closeEvents = await Promise.all([
+                    new Promise<CloseEvent>(resolve => client1.addEventListener('close', resolve)),
+                    new Promise<CloseEvent>(resolve => client2.addEventListener('close', resolve)),
+                ])
+                expect(service.connectionCount).toBe(0)
+
+                closeEvents.forEach(evt => {
+                    expect(evt.code).toEqual(1000)
+                    expect(evt.reason).toEqual('')
+                })
```

Comment:
@ilikejames I just pushed the implementation and tests on the new version.

I've added an assertion to the default behaviour (calling `.disconnect()`), and another one down below passing in an explicit option.

I've set the defaults to [1000, ""] because I've tested this in master and one host defaults to [1000, ""] and the other to [1005, ""]. The spec says:

>    1005 is a reserved value and MUST NOT be set as a status code in a
>    Close control frame by an endpoint.  It is designated for use in
>    applications expecting a status code to indicate that no status
>    code was actually present.

So I went with the other default.

Reply:
Let me guess, the client of the nodejs service one results in 1005, because by default there was not one sent? 😊

I think theres going to be quite a few more changes to be made to get the build green. 🪱🪱🪱

Mind if I take a look?

Reply:
Sorry, I'll open them 😄 

I think the `wasClean` prop is something that leaks out from mock-socket, but it's unused when calling `.close(options)`. They are just reusing the interface for both the CloseEvents they create (because that one needs wasClean as part of the spec) with the options that the function `.close()` takes (which no spec takes a wasClean, they only take close and reason).

The flow of that goes like this:
1. `.close()` calls the proxy into this line: https://github.com/thoov/mock-socket/blob/master/src/helpers/proxy-factory.js#L15C4-L15C4
2. This calls `closeWebSocketConnection` only passing `code` and `reason` => https://github.com/thoov/mock-socket/blob/master/src/algorithms/close.js#L6
3. `closeWebSocketConnection` calls `createCloseEvent`, and that generates the CloseEvent without passing in `wasClean`, and that just takes the default value

`wasClean` is only passed as false on fail case, which is an alternate flow on the client's WebSocket `.close(code, reason)` https://github.com/thoov/mock-socket/blob/master/src/websocket.js#L186C14-L186C14 if the connection wasn't established yet.

This is why I opted to ignore it, sorry I didn't add any comment trying to explain this.... I would either ask mock-socket to fix that TS interface, or hack it with `as any` or just giving it whatever value (I did the last option I could think of)

Reply:
Thanks for the deep dive! 

I was actually referring to the first hash i provided. But it was late for me, and I'd had a long work day so I probably wasn't very clear. 

The 2nd hash is good I think. Either cherrypick it, or use it for your own basis. 

Have a good weekend!

Reply:
Hey @voliva want to move forward with this? Also, I sent you a `X` dm. Well wishes.

Reply:
Hey sorry, I was on holidays.

I cherry picked the second hash, that looks really good, thank you! I've only removed an unused variable on `testSetup.ts`